sshagent_windows.go

```diff
@@ -22,22 +22,46 @@
 package sshagent
 
 import (
-	"errors"
+	"fmt"
 	"io"
 	"net"
 	"sync"
 
+	"github.com/Microsoft/go-winio"
 	"golang.org/x/crypto/ssh/agent"
 )
 
+const (
+	sshAgentPipe = `\\.\pipe\openssh-ssh-agent`
+)
+
+// Available returns true if Pageant is running
+func Available() bool {
+	if pageantWindow() != 0 {
+		return true
+	}
+	conn, err := winio.DialPipe(sshAgentPipe, nil)
+	if err != nil {
+		return false
+	}
+	conn.Close()
+	return true
+}
+
 // New returns a new agent.Agent and the (custom) connection it uses
 // to communicate with a running pagent.exe instance (see README.md)
 func New() (agent.Agent, net.Conn, error) {
-	if !Available() {
-		return nil, nil, errors.New("SSH agent requested but Pageant not running")
+	if pageantWindow() != 0 {
+		return agent.NewClient(&conn{}), nil, nil
 	}
-
-	return agent.NewClient(&conn{}), nil, nil
+	conn, err := winio.DialPipe(sshAgentPipe, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf(
+			"SSH agent requested but Pageant not running and Error %v",
+			err,
+		)
+	}
+	return agent.NewClient(conn), nil, nil
```

Comment:
I wonder if we should make the Windows Native agent the default (i.e. test that one first). What do you think? Also I think we need to make the error more clear in that it tested for both Pageant and the native SSH agent and that neither are available.

Reply:
> I wonder if we should make the Windows Native agent the default (i.e. test that one first). What do you think?

Considering that only Pageant was supported originally, I think the compatibility is better if you make Pageant the default.

> Also I think we need to make the error more clear in that it tested for both Pageant and the native SSH agent and that neither are available.

For example, like following?

```
SSH agent requested but Pageant not running and cannot connect native SSH agent.
```

Reply:
> Considering that only Pageant was supported originally, I think the compatibility is better if you make Pageant the default.

Agree...

Or something like:
```
SSH agent requested, but could not detect Pageant or Windows native SSH agent.
```